adminSite/client/ChartEditor.ts

```diff
@@ -165,38 +165,46 @@ export class ChartEditor {
         )
     }
 
-    async saveChart({ onError }: { onError?: () => void } = {}) {
-        const { chart, isNewChart, currentChartJson } = this
+    async saveGrapher({ onError }: { onError?: () => void } = {}) {
+        const { grapher, isNewGrapher, currentGrapherObject } = this
+
+        // Chart title and slug may be autocalculated from data, in which case they won't be in props
+        // But the server will need to know what we calculated in order to do its job
+        if (!currentGrapherObject.title)
+            currentGrapherObject.title = grapher.displayTitle
+
+        if (!currentGrapherObject.slug)
+            currentGrapherObject.slug = grapher.displaySlug
```

Comment:
It would be nice if we could ditch that. I think we need it when saving though (so all the site code like the Chart Index page can be sure to have a title, since those places aren't rehydrating the Grapher instance, just going off the JSON).

I wouldn't be surprised if I missed a case here though and there's a regression. In which case we could add a test and also thus document it's usage.